imports.py

```diff
@@ -12,193 +10,283 @@
-	if 'issue' in response:
-		print("error")
-	else:
-		for p in response['parameter']:
-			if p['name'] != 'property' and p['name'] != 'designation':
-				if 'valueString' in p:
-					print(p['name'], ":", p['valueString'], sep="", end=", ")
-				if 'valueBoolean' in p:
-					print(p['name'], ":", p['valueBoolean'], sep="", end=", ")
-		print("")
 
 
-class TimsClient:
-	"""TIMS Client"""
-	http_client = httpx.Client()
-
-	def __init__(self, base_url: str, use_cache=False):
-		self.base_url = base_url
-		self.use_cache = use_cache
-
-	def do_hapi_request(self, internal_path: str, in_params: Dict = None) -> Dict:
-		"""GET request"""
-		url = os.path.join(self.base_url, internal_path)
-		if self.use_cache:
-			response_dict = cache_read(url, in_params)
-			if response_dict:
-				return response_dict
-		rq = httpx.Request("GET", url, params=in_params, headers=REQUEST_HEADERS)
-		response: Response = TimsClient.http_client.send(rq)
-		response_dict: Dict = json.loads(response.text)
-
-		cache_write(url, in_params, response_dict)
-		return response_dict
-
-	def summarize_code_systems(self, verbose=False):
-		"""Summarize code systems"""
-		response_dict = self.do_hapi_request("CodeSystem?_summary=true", {})
-		# if not decode_response_status_ok(response):
-		# 	response_dict = json.loads(response.text)
-		if verbose:
-			for vocab_entry in response_dict['entry']:
-				print(vocab_entry['resource']['name'])
-				print(vocab_entry['resource']['id'])
-				print(vocab_entry['fullUrl'])
-				print(vocab_entry['resource']['url'])
-				print('')
-		else:
-			sys_names = [x['resource']['name'] for x in response_dict['entry']]
-			print(', '.join(sys_names))
-
-	def summarize_code_system_by_name(self, name: str):
-		"""Summarize a code system by name param"""
-		response: Dict = self.do_hapi_request("CodeSystem", {'name': name})
-		# if not decode_response_status_not_ok(response):
-		decode_code_system_json(response)
-
-	def summarize_code_system_by_system(self, system: str):
-		"""Summarize a code system by system param"""
-		response: Dict = self.do_hapi_request("CodeSystem", {'system': system})
-		# if not decode_response_status_not_ok(response):
-		decode_code_system_json(response)
-
-	def validate_code(self, system_id: str, code: str):
-		"""Validate that code is in code system"""
-		response: Dict = self.do_hapi_request(f'CodeSystem/{system_id}/$validate-code?code={code}')
-		valid = [x['valueBoolean'] for x in response['parameter'] if x['name'] == 'result'][0]
-		if valid:
-			label = [x['valueString'] for x in response['parameter'] if x['name'] == 'display'][0]
-			print(label)
-		else:
-			print('Invalid code')
-
-	def lookup_code(self, system: URI, code: str):
-		"""Lookup code within code system"""
-		response: Dict = self.do_hapi_request(f'CodeSystem/$lookup?system={system}&code={code}')
-		decode_lookup_response(response)
+class CacheFhir:
+    """ missing an invalidation mechanism, use with caution """
```

Comment:
Since you opened the OO here, I thought this made sense. If only to group them.

Reply:
Also, I see why you set this up. The JHU server, or the net between here and there, isn't nearly as fast as a local server. Local just takes a few milli seconds. JHU could be 10 seconds. I hope the memory helps it out.